include/envoy/http/filter.h

```diff
@@ -221,9 +221,18 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    *                  type, or encoded in the grpc-message header.
    * @param modify_headers supplies an optional callback function that can modify the
    *                       response headers.
+   * @param rate_limited_as_resource_exhausted specifies whether a RESOURCE_EXHAUSTED code
+   *                                           should be returned instead of the default
+   *                                           UNAVAILABLE code for rate limited gRPC calls.
    */
   virtual void sendLocalReply(Code response_code, const std::string& body_text,
-                              std::function<void(HeaderMap& headers)> modify_headers) PURE;
+                              std::function<void(HeaderMap& headers)> modify_headers,
```

Comment:
I don't rate-limit specific bool config flag should be leaked into HTTP filter interface (and HttpUtility etc). We can make this interface containing an HTTP status and an `absl::optional` of gRPC status. As HTTP code to gRPC code is not 1:1, so this could be used to address #3619 in future as well. cc @jrajahalme 

Alternatively you can make this change without modifying the interface here, just change grpc-status header in `modify_headers` callback passed by ratelimit filter.

Reply:
Agree with @lizan . It should not be leaked to Http Filter interface.

Reply:
I've addressed this in b7cd9e7. The filter/utility APIs now take a `absl::optional<Grpc::StatusMap>` input with the HTTP to gRPC code mapping.

Reply:
Why do you need a map here? Callers of `sendLocalReply` knows what response status is, so just an `absl::optional<GrpcStatus>` would suffice. i.e. if the supplied grpc status was nullopt, then use default map. If it exist, just use the supplied grpc status.

Reply:
Oops, fixed it in bd1201e.